jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/Collections.java

```diff
@@ -1848,7 +1848,7 @@ public boolean equals(Object o) {
 
         SynchronizedCollection(Collection<E> c) {
             this.c = Objects.requireNonNull(c);
-            mutex = this;
+            mutex = new Object();
```

Comment:
Nice fix. Could you please branch this file over to ojluni/src/objc. Our convention is for modified sources to be moved from src/main to src/objc. Sorry for the extra work.